service/common/src/main/java/org/apache/polaris/service/catalog/policy/PolicyCatalog.java

```diff
@@ -251,25 +248,224 @@ public Policy updatePolicy(
+    LoadPolicyMappingsResult result;
+    if (policyType == null) {
+      result = metaStoreManager.loadPoliciesOnEntity(callContext.getPolarisCallContext(), target);
+    } else {
+      result =
+          metaStoreManager.loadPoliciesOnEntityByType(
+              callContext.getPolarisCallContext(), target, policyType);
+    }
+
+    return result.getEntities().stream().map(PolicyCatalog::toPolicy).toList();
+  }
+
+  private List<PolarisEntity> getFullPath(Namespace namespace, String targetName) {
+    if (namespace == null || namespace.isEmpty()) {
+      // catalog
+      return List.of(catalogEntity);
+    } else if (Strings.isNullOrEmpty(targetName)) {
+      // namespace
+      var resolvedTargetEntity = resolvedEntityView.getResolvedPath(namespace);
+      if (resolvedTargetEntity == null) {
+        throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+      }
+      return resolvedTargetEntity.getRawFullPath();
+    } else {
+      // table
+      var tableIdentifier = TableIdentifier.of(namespace, targetName);
+      // only Iceberg tables are supported
+      var resolvedTableEntity =
+          resolvedEntityView.getResolvedPath(
+              tableIdentifier, PolarisEntityType.TABLE_LIKE, PolarisEntitySubType.ICEBERG_TABLE);
+      if (resolvedTableEntity == null) {
+        throw new NoSuchTableException("Iceberg Table does not exist: %s", tableIdentifier);
+      }
+      return resolvedTableEntity.getRawFullPath();
+    }
+  }
+
+  private PolarisResolvedPathWrapper getResolvedPathWrapper(PolicyIdentifier policyIdentifier) {
+    var resolvedEntities =
         resolvedEntityView.getPassthroughResolvedPath(
             policyIdentifier, PolarisEntityType.POLICY, PolarisEntitySubType.NULL_SUBTYPE);
-    if (resolvedEntities == null) {
+    if (resolvedEntities == null || resolvedEntities.getResolvedLeafEntity() == null) {
       throw new NoSuchPolicyException(String.format("Policy does not exist: %s", policyIdentifier));
     }
+    return resolvedEntities;
+  }
 
-    List<PolarisEntity> catalogPath = resolvedEntities.getRawParentPath();
-    PolarisEntity leafEntity = resolvedEntities.getRawLeafEntity();
-
-    DropEntityResult dropEntityResult =
-        metaStoreManager.dropEntityIfExists(
-            callContext.getPolarisCallContext(),
-            PolarisEntity.toCoreList(catalogPath),
-            leafEntity,
-            Map.of(),
-            false);
+  private PolarisResolvedPathWrapper getResolvedPathWrapper(
+      @Nonnull PolicyAttachmentTarget target) {
+    return switch (target.getType()) {
+      // get the current catalog entity, since policy cannot apply across catalog at this moment
+      case CATALOG -> resolvedEntityView.getResolvedReferenceCatalogEntity();
+      case NAMESPACE -> {
+        var namespace = Namespace.of(target.getPath().toArray(new String[0]));
+        var resolvedTargetEntity = resolvedEntityView.getResolvedPath(namespace);
```

Comment:
Where do these paths get added to the resolver?

Reply:
IIUC, we will need to create a `PolicyCatalog` object [here](https://github.com/polaris-catalog/polaris/blob/main/service/common/src/main/java/org/apache/polaris/service/context/PolarisCallContextCatalogFactory.java#L86-L86), just like what we did for `IcebergCatalog`. cc @HonahX as he created this class in #1294.